src/Tasks/Microsoft.NET.Build.Tasks/GenerateSupportedTargetFrameworkAlias.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using NuGet.Frameworks;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    public class GenerateSupportedTargetFrameworkAlias : TaskBase
+    {
+        [Required]
+        public ITaskItem[] SupportedTargetFramework { get; set; }
+
+        [Required]
+        public string TargetFrameworkMoniker { get; set; }
+
+        public string TargetPlatformMoniker { get; set; }
```

Comment:
I'm not following how this design is supposed to work.  The list of items in the dropdown shouldn't depend on what the current target framework is, so I don't think this task should be taking the current target framework and platform as parameters.

I think it should basically take UseWpf and UseWindowsForms as arguments, and if either of them is true, it should remove any supported frameworks below .NET Core 3.0, and add "-windows" to any of them that are 5.0 or higher.

Reply:
Currently the target framework drop down is bound to the `TargetFrameworkMoniker`, so part of this change is changing it to bind to the `TargetFramework` instead (see [this line](https://github.com/dotnet/sdk/pull/13545/files/1d6963e324995cee06ccdae90325a54c146e8eb9#diff-11e7b7431abbd0596f030eefc10620e9R42)). 

For generality, I'm constructing the target framework from both the `TargetFrameworkMoniker` and `TargetPlatformMoniker` values. This requires the special case [here](https://github.com/dotnet/sdk/pull/13545/files/1d6963e324995cee06ccdae90325a54c146e8eb9#diff-11e7b7431abbd0596f030eefc10620e9R33) because the `TargetPlatformMonkier` defaulted to windows 7.0 before 5.0, so we want to remove that unless we really are targeting windows (that is, UseWPF or UseWindowsForms are set).

Reply:
What are the valid `TargetFramework` values for a Windows Forms or WPF project?  It should be this:

- `netcoreapp3.0`
- `netcoreapp3.1`
- `net5.0-windows`

That list doesn't depend on what the current target framework or platform is.  So I would write the code something like this:

```c#
foreach (var tfm in SupportedTargetFramework)
{
    var targetPlatform = "";
    var targetFramework = NuGetFramework.Parse(tfm.ItemSpec);
    if (UseWindowsForms || UseWpf)
    {
        if (targetFramework.Version.Major >= 5)
        {
            targetFramework.Platform = "Windows";
        }
        if (targetFramework.Version.Major < 3)
        {
            //  Windows Forms / WPF not supported below .NET Core 3.0
            continue;
        }
    }
    var targetFrameworkAlias = targetFramework.GetShortFolderName();
    var displayName = string.IsNullOrWhiteSpace(tfm.GetMetadata("DisplayName"))? targetFrameworkAlias : tfm.GetMetadata("DisplayName");
    convertedTfms.Add(new TaskItem(targetFrameworkAlias, new Dictionary<string, string>() { { "DisplayName", displayName } }));
}
```

(This code doesn't account for projects that target something besides .NETCoreApp, so that would also need to be taken into consideration).

Reply:
My motivation for using the target platform is generality in the future so that we're minimizing the special casing for windows. This way the code works for target platforms besides windows. 

The platform property in a nuget framework is also readonly, so we'll need a target platform moniker to use in initialization to set the platform property regardless.

Reply:
@dsplaisted per our discussion I've added the test case you suggested and made the code more explicit. Let me know if you have any other feedback.